substratest/factory.py

```diff
@@ -402,162 +397,86 @@ def __init__(self, dataset) -> None:
         )
 
 
-class ComputePlanTraintupleSpec(substra.sdk.schemas.ComputePlanTraintupleSpec):
-    @property
-    def id(self):
-        return self.traintuple_id
-
-
-class ComputePlanAggregatetupleSpec(substra.sdk.schemas.ComputePlanAggregatetupleSpec):
-    @property
-    def id(self):
-        return self.aggregatetuple_id
-
-
-class ComputePlanCompositeTraintupleSpec(substra.sdk.schemas.ComputePlanCompositeTraintupleSpec):
-    @property
-    def id(self):
-        return self.composite_traintuple_id
-
-
-# All of those will be removed when inputs consumers will be merge
-def _get_data_manager_from_inputs(inputs):
-    data_manager_keys = [inp.asset_key for inp in inputs if inp.identifier == InputIdentifiers.opener]
-    assert len(data_manager_keys) == 1
-    return data_manager_keys[0]
-
-
-def _get_data_samples_from_inputs(inputs):
-    data_samples = [inp.asset_key for inp in inputs if inp.identifier == InputIdentifiers.datasamples]
-    return data_samples
-
-
-def _get_in_models_from_inputs(inputs):
-    in_models = [
-        t.parent_task_key
-        for t in inputs
-        if t.identifier in (InputIdentifiers.models, InputIdentifiers.model, InputIdentifiers.shared)
-    ]
-    return in_models
-
-
-def _get_head_trunk_model_from_inputs(inputs):
-    in_head_model_key = [t.parent_task_key for t in inputs if t.identifier == InputIdentifiers.local]
-    in_trunk_model_key = [t.parent_task_key for t in inputs if t.identifier == InputIdentifiers.shared]
-    assert len(in_head_model_key) <= 1
-    assert len(in_trunk_model_key) <= 1
-    in_head_model_key = in_head_model_key[0] if len(in_head_model_key) == 1 else None
-    in_trunk_model_key = in_trunk_model_key[0] if len(in_trunk_model_key) == 1 else None
-    assert type(in_head_model_key) == type(in_trunk_model_key)
-
-    return in_head_model_key, in_trunk_model_key
-
-
-def _get_predict_tuple_from_inputs_for_testtuple(inputs):
-    predictions = [t.parent_task_key for t in inputs if t.identifier == InputIdentifiers.predictions]
-    assert len(predictions) == 1
-    return predictions[0]
-
-
-def _get_train_tuple_from_inputs_for_predicttuple(inputs):
-    traintuple_ids = set(t.parent_task_key for t in inputs if t.parent_task_key is not None)
-
-    assert len(traintuple_ids) == 1
-    traintuple_id = traintuple_ids.pop()
-
-    return traintuple_id
-
-
 class _ComputePlanSpecFactory:
-    def create_traintuple(self, algo, inputs=None, outputs=None, tag="", metadata=None) -> ComputePlanTraintupleSpec:
+    def create_traintuple(self, algo, worker, inputs=None, outputs=None, tag="", metadata=None) -> ComputePlanTaskSpec:
```

Comment:
Maybe these methods are not so much necessary anymore. We can define default outputs upstream and use a generic create_task method. But this could be easily done later.

Reply:
As you say, we should review our whole test process for generic task, some tests are now redundant
These methods are useful to have the right inputs/outputs for each task, and connect depending tasks correctly for now